backend-lib/Cargo.toml

```diff
@@ -63,10 +63,10 @@ crate-type = ["staticlib", "cdylib"]
 [patch.crates-io]
 incrementalmerkletree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "337f59179eda51261e9ddfc6b18e8fb84ea277c9" }
 shardtree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "337f59179eda51261e9ddfc6b18e8fb84ea277c9" }
-zcash_address = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_client_backend = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_client_sqlite = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_encoding = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_primitives = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_proofs = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_protocol = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
+zcash_address = { git = "https://github.com/zcash/librustzcash.git", rev = "eaa43b4e2597786317e59863e02bf0701152f398" }
```

Comment:
There are necessary bugfixes in the upstream Rust crates, but given that #1533 also alters these revisions, I propose that we merge this PR and #1533, and then I'll make another PR that updates the feature branch with bugfixes.